### PR TITLE
CEDS-1851 Items summary page allows a Simplified journey item

### PR DESCRIPTION
**Pull request by PauloPortugal**
As part of the Simplified journey, we had to remove the commodity
measurements page from the item journey.
This change implies a change on the Items Summary validation rules for
the journey in question.

### CODE REVIEW

File: app/controllers/declaration/ItemsSummaryController.scala [hunk 1]
```diff
@@ -67,8 +68,8 @@ class ItemsSummaryController @Inject()(
     }
   }
 
-  private def buildIncorrectItemsErrors(items: Seq[ExportItem]): Seq[FormError] =
-    items.zipWithIndex.filterNot { case (item, _) => item.isCompleted }.map {
+  private def buildIncorrectItemsErrors(request: JourneyRequest[_]): Seq[FormError] =
```
Comment (PatrykRudnicki) [inline app/controllers/declaration/ItemsSummaryController.scala]: I think it's better to use `JourneyRequest[AnyContent]` than `[_]`

File: app/models/declaration/ExportItem.scala [hunk 2]
```diff
@@ -35,9 +37,15 @@ case class ExportItem(
...
-    procedureCodes.isDefined && isFiscalInformationCompleted && itemType.isDefined &&
-      packageInformation.nonEmpty && commodityMeasure.isDefined && additionalInformation.isDefined
+  val isCompleted: PartialFunction[DeclarationType, Boolean] = {
+    case DeclarationType.STANDARD | DeclarationType.SUPPLEMENTARY =>
+      procedureCodes.isDefined && isFiscalInformationCompleted && itemType.isDefined &&
+        packageInformation.nonEmpty && commodityMeasure.isDefined && additionalInformation.isDefined
+
+    case DeclarationType.SIMPLIFIED =>
+      procedureCodes.isDefined && isFiscalInformationCompleted && itemType.isDefined &&
+        packageInformation.nonEmpty && additionalInformation.isDefined
```
Comment (PatrykRudnicki) [inline app/models/declaration/ExportItem.scala]: If we extract validation to method like `supplementaryDecValidation` or some kind `baseValidation` we can reuse it for Standard and Supplementary like `baseValidation && commodityMeasure.isDefined`.

Non blocking comment, I'm not sure about naming here and the value.

File: app/controllers/declaration/ItemsSummaryController.scala (hunk 1 above)
Comment (PauloPortugal) [reply app/controllers/declaration/ItemsSummaryController.scala]: Very good point! :)

Comment (bambuchaAdm) [reply app/controllers/declaration/ItemsSummaryController.scala]: Here you just does not matter about `request.body` so `[_]` is more "any content" then `Request[AnyContent]` is.

File: app/models/declaration/ExportItem.scala (hunk 2 above)
Comment (bambuchaAdm) [reply app/models/declaration/ExportItem.scala]: Generally i would not share validation rules here between journeys  - even if they looks similar

**Pull request merged**